### ISSUE TITLE
question regarding unfamiliarity

### ISSUE THREAD

**Issue by orgw** [molML/JointMolecularModel#1]
Hi guys, first off awesome work.

one question regarding a approach i tried. I used a massive pretrained encoder-decoder(transformer based, 91 million SMILES samples sourced from PubChem,) and it seems the reconstruction loss, here we use cross entropy (averaged by length) of test and OOD sets are nearly identical. i suppose this happens when the model is massively trained but i wonder how we would get a unfamiliarity metric in this case. maybe i should train from scratch with only 

thanks

below is a plot comparing frequency(y-axis) and loss(x-axis). as you can see most ood molecules are reconstructed appropriately.

![Image](https://github.com/user-attachments/assets/565f2403-9c16-4206-ab31-74b509904e5a)

**Comment by derekvantilborg** [comment]
Hi, it might indeed be a problem if you're reconstructing everything perfectly (see Zhou et al. (CVPR 2022), “Rethinking Reconstruction Autoencoder-Based Out-of-Distribution Detection”) and you can try regularizing your model a bit more for example. However, its probably best to retrain from scratch indeed. In our paper we make sure that the pretraining data is not too similar to the finetuning data so that our models are not too biased to that chemical matter. But at the end of the day, the pretraining is just to get the AE goin, as those are difficult to train with tiny datasets. I suspect that you don't need the perfectly trained encoder-decoder anyways and as long as your encoder-decoder functions reasonably its fine I think.
